File: OnlineStore/Api/CommonEntities/Additional/OrderStatus.cs
using System;


namespace CommonEntities.Additional
{
	public class OrderStatus
	{
		public const string StatusNewOrder = "NewOrder";
		public const string StatusProcessed = "Processed";
		public const string StatusPaid = "Paid";
		public const string StatusWaitingForDelivery = "WaitingForDelivery";
		public const string StatusCanceled = "Canceled";
		public const string StatusFulfilled = "Fulfilled";

		public enum OrderStatusEnum
		{
			NewOrder = 10,
			Processed = 11,
			Paid = 12,
			WaitingForDelivery =13,
			Canceled =14,
			Fulfilled = 15
		}


		public OrderStatusEnum Status { get; }

		public OrderStatus(string status)
		{
			Status = GetStatus(status);
		}
		public OrderStatus(OrderStatusEnum status)
		{
			Status = status;
		}


		public string GetStatusName()
		{
			return GetStatusName(Status);
		}


		public static string GetStatusName(OrderStatusEnum status)
		{
			switch (status)
			{
				case OrderStatusEnum.NewOrder: return StatusNewOrder;
				case OrderStatusEnum.Processed: return StatusProcessed;
				case OrderStatusEnum.Paid: return StatusPaid;
				case OrderStatusEnum.WaitingForDelivery: return StatusWaitingForDelivery;
				case OrderStatusEnum.Fulfilled: return StatusFulfilled;
				case OrderStatusEnum.Canceled: return StatusCanceled;
				default: throw new ArgumentOutOfRangeException(nameof(status), status, null);
			}
		}
		public static OrderStatusEnum GetStatus(string status)
		{
			switch (status)
			{
				case StatusNewOrder: return OrderStatusEnum.NewOrder;
				case StatusProcessed: return OrderStatusEnum.Processed;
				case StatusPaid: return OrderStatusEnum.Paid;
				case StatusWaitingForDelivery: return OrderStatusEnum.WaitingForDelivery;
				case StatusFulfilled: return OrderStatusEnum.Fulfilled;
				case StatusCanceled: return OrderStatusEnum.Canceled;
				default: throw new ArgumentOutOfRangeException(nameof(status), status, null);
			}
		}


	}
}

File: OnlineStore/OnlineStoreUI/src/model/entities/apiEntities/additional/orderStatus.ts
export class OrderStatus {
    status:OrderStatusEnum;

    constructor(status:OrderStatusEnum){
        this.status = status;
    }
   
    static getStatusPrint(status:OrderStatusEnum){

        if(status == OrderStatusEnum.NewOrder)
            return 'New order';
        if(status == OrderStatusEnum.Processed)
            return 'Processed';
        if(status == OrderStatusEnum.Paid)
            return 'Paid';
        if(status == OrderStatusEnum.WaitingForDelivery)
            return 'Waiting for delivery';
        if(status == OrderStatusEnum.Canceled)
            return 'Canceled';
        if(status == OrderStatusEnum.Fulfilled)
            return 'Fulfilled';
        
        return '';
    }

    static getAllOrderStatus():OrderStatus[]{

        let allStatus: OrderStatus[] = [
            new OrderStatus(OrderStatusEnum.NewOrder),
            new OrderStatus(OrderStatusEnum.Processed),
            new OrderStatus(OrderStatusEnum.Paid),
            new OrderStatus(OrderStatusEnum.WaitingForDelivery),
            new OrderStatus(OrderStatusEnum.Canceled),
            new OrderStatus(OrderStatusEnum.Fulfilled)
        ]

        return allStatus;
    }

    static getAllOrderStatusEnum():OrderStatusEnum[]{

        let allStatus: OrderStatusEnum[] = [
            OrderStatusEnum.NewOrder,
            OrderStatusEnum.Processed,
            OrderStatusEnum.Paid,
            OrderStatusEnum.WaitingForDelivery,
            OrderStatusEnum.Canceled,
            OrderStatusEnum.Fulfilled
        ]

        return allStatus;
    }

}

export enum OrderStatusEnum {
    NewOrder = 10,
	Processed = 11,
	Paid = 12,
	WaitingForDelivery = 13,
	Canceled = 14,
	Fulfilled = 15
}